src/BrazeKit-dev.js

```diff
@@ -709,6 +719,108 @@ var constructor = function () {
         }
     }
 
+    function prepareInitialConsent(user) {
+        var userConsentState = getUserConsentState(user);
+
+        var currentConsentPayload = generateBrazeConsentStatePayload(
+            userConsentState
+        );
+
+        latestUserBrazeConsentString = JSON.stringify(currentConsentPayload);
+
+        if (latestUserBrazeConsentString.length) {
+            setConsentOnBraze(currentConsentPayload);
+        }
+    }
+
+    function setConsentOnBraze(currentConsentPayload) {
+        for (var key in currentConsentPayload) {
+            braze
+                .getUser()
+                .setCustomUserAttribute(key, currentConsentPayload[key]);
+        }
+    }
+
+    function maybeSetConsentBeforeEventLogged(event) {
+        if (latestUserBrazeConsentString && parsedConsentMappings.length) {
+            var eventConsentState = getEventConsentState(event.ConsentState);
+
+            if (!isEmpty(eventConsentState)) {
+                var eventBrazeConsent = generateBrazeConsentStatePayload(
+                    eventConsentState
+                );
+                var eventBrazeConsentAsString = JSON.stringify(
+                    eventBrazeConsent
+                );
+
+                if (
+                    eventBrazeConsentAsString !== latestUserBrazeConsentString
+                ) {
+                    setConsentOnBraze(eventBrazeConsent);
+                    latestUserBrazeConsentString = eventBrazeConsentAsString;
+                }
+            }
+        }
+    }
+
+    function getEventConsentState(eventConsentState) {
+        return eventConsentState && eventConsentState.getGDPRConsentState
+            ? eventConsentState.getGDPRConsentState()
+            : {};
+    }
+
+    function generateBrazeConsentStatePayload(consentState) {
+        if (!parsedConsentMappings) return {};
+
+        var payload = {};
+
+        // These are Braze's consent constants for Braze's Audience Sync to Google
+        // https://www.braze.com/docs/partners/canvas_steps/google_audience_sync
+
+        var googleToBrazeConsentMap = {
+            google_ad_user_data: '$google_ad_user_data',
+            google_ad_personalization: '$google_ad_personalization',
+        };
```

Comment:
I think this is repetitive code and can be removed? we have the same constants in an array at line 77